packages/dataviews/src/components/dataviews-view-config/index.tsx

```diff
@@ -1,150 +1,195 @@
 /**
  * External dependencies
  */
-import type { ChangeEvent } from 'react';
+import clsx from 'clsx';
 
 /**
  * WordPress dependencies
  */
 import {
 	Button,
-	privateApis as componentsPrivateApis,
+	Popover,
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	SelectControl,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalItem as Item,
+	__experimentalGrid as Grid,
+	__experimentalVStack as VStack,
 	__experimentalHStack as HStack,
 } from '@wordpress/components';
 import { __, _x } from '@wordpress/i18n';
-import { memo, useContext } from '@wordpress/element';
-import { cog } from '@wordpress/icons';
+import { memo, useContext, useState, useMemo } from '@wordpress/element';
+import { cog, seen } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
-import { unlock } from '../../lock-unlock';
-import { SORTING_DIRECTIONS, sortLabels } from '../../constants';
+import { SORTING_DIRECTIONS, sortLabelsShort } from '../../constants';
 import { VIEW_LAYOUTS, getMandatoryFields } from '../../layouts';
-import type { NormalizedField, View, SupportedLayouts } from '../../types';
+import type { SupportedLayouts } from '../../types';
 import DataViewsContext from '../dataviews-context';
 
-const {
-	DropdownMenuV2: DropdownMenu,
-	DropdownMenuGroupV2: DropdownMenuGroup,
-	DropdownMenuItemV2: DropdownMenuItem,
-	DropdownMenuRadioItemV2: DropdownMenuRadioItem,
-	DropdownMenuCheckboxItemV2: DropdownMenuCheckboxItem,
-	DropdownMenuItemLabelV2: DropdownMenuItemLabel,
-} = unlock( componentsPrivateApis );
-
-interface ViewTypeMenuProps {
-	view: View;
-	onChangeView: ( view: View ) => void;
+interface ViewActionsProps {
 	defaultLayouts?: SupportedLayouts;
 }
 
-interface PageSizeMenuProps {
-	view: View;
-	onChangeView: ( view: View ) => void;
-}
-
-interface FieldsVisibilityMenuProps< Item > {
-	view: View;
-	onChangeView: ( view: View ) => void;
-	fields: NormalizedField< Item >[];
+function LayoutPicker( {
+	defaultLayouts = { list: {}, grid: {}, table: {} },
+}: ViewActionsProps ) {
+	const { view, onChangeView } = useContext( DataViewsContext );
+	const availableLayouts = Object.keys( defaultLayouts );
+	return (
+		<ToggleGroupControl
+			__nextHasNoMarginBottom
+			__next40pxDefaultSize
+			isBlock
+			label={ __( 'Layout' ) }
+			value={ view.type }
+			onChange={ ( newLayout ) => {
+				switch ( newLayout ) {
+					case 'list':
+					case 'grid':
+					case 'table':
+						return onChangeView( {
+							...view,
+							type: newLayout,
+							...defaultLayouts[ newLayout ],
+						} );
+				}
+				throw new Error( 'Invalid dataview' );
```

Comment:
Do we really want to throw errors in an `onChange` handler? I'd expect a `warning()` or something less invasive. I see this was here before, so just something to keep in mind.

Reply:
Good point, I followed what we had without thinking much but a warning seems like a better option. The code was updated 👍